### PR TITLE
lib: nrf_modem: trace: add option to pause and resume uart trace backend

### PR DESCRIPTION
**Pull request by eivindj-nordic**
Add option to pause and resume UART trace backend to make it more power efficient.

### CODE REVIEW

**Comment by junqingzou** [comment]
After disable HWFC in DTS `modem-trace-uart.overlay`, verified PSM sleep current to be 3.3uA

**Review by junqingzou — APPROVED** [review]

**Review by lemrey — APPROVED** [review]
👍🏻

**Review by divipillai — APPROVED** [review]

File: doc/nrf/libraries/modem/nrf_modem_lib/nrf_modem_lib_trace.rst [hunk 1]
```diff
@@ -238,13 +238,33 @@ Complete the following steps to add a custom trace backend:
           */
       }
 
+      int trace_backend_suspend(void)
+      {
+         /* This function allows the trace module to suspend the trace backend. When suspended,
+          * the backend cannot be used by the trace module until it is resumed by calling
+          * trace_backend_resume().
```
Comment (MirkoCovizzi) [inline doc/nrf/libraries/modem/nrf_modem_lib/nrf_modem_lib_trace.rst:245]: Nit:
```suggestion
          * `trace_backend_resume()`.
```

File: doc/nrf/libraries/modem/nrf_modem_lib/nrf_modem_lib_trace.rst [hunk 2]
```diff
@@ -238,13 +238,33 @@ Complete the following steps to add a custom trace backend:
...
+          *
+          * If not applicable for the trace backend, set to NULL in the trace_backend struct.
```
Comment (MirkoCovizzi) [inline doc/nrf/libraries/modem/nrf_modem_lib/nrf_modem_lib_trace.rst:247]: Nit:
```suggestion
          * If not applicable for the trace backend, set to NULL in the `trace_backend` struct.
```

File: doc/nrf/libraries/modem/nrf_modem_lib/nrf_modem_lib_trace.rst [hunk 3]
```diff
@@ -238,13 +238,33 @@ Complete the following steps to add a custom trace backend:
...
+          */
+      }
+
+      int trace_backend_resume(void)
+      {
+         /* This function allows the trace module to resume the trace backend after it is suspended.
+          *
+          * If not applicable for the trace backend, set to NULL in the trace_backend struct.
```
Comment (MirkoCovizzi) [inline doc/nrf/libraries/modem/nrf_modem_lib/nrf_modem_lib_trace.rst:255]: Nit:
```suggestion
          * If not applicable for the trace backend, set to NULL in the `trace_backend` struct.
```

**Review by MirkoCovizzi — COMMENTED** [review]
Nice! Just left some nits

**Review by MirkoCovizzi — APPROVED** [review]

**Pull request merged**